### DISCUSSION TITLE
KeyError: 'create_route'

### DISCUSSION THREAD

**Discussion by dkharyna** [miguelgrinberg/python-socketio#778] [Q&A]
Does it correct exception, when I try to use python-socketio with Quart?

> File "C:\Users\a\ce-svc\src\.\app\app.py", line 60, in <module>
    sio.attach(app)
  File "C:\Users\a\ce-svc\env\lib\site-packages\socketio\asyncio_server.py", line 111, in attach
    self.eio.attach(app, socketio_path)
  File "C:\Users\a\ce-svc\env\lib\site-packages\engineio\asyncio_server.py", line 78, in attach
    self._async['create_route'](app, self, '/{}/'.format(engineio_path))
KeyError: 'create_route'

I use AsyncServer with async_mode=asgi

**Comment by miguelgrinberg** [comment]
Can I ask you to not write issues when all you have are questions? Issues are for bugs that you have found in this package. If you just have a question, or you don't know if what you have is a bug, then post to the discussions board to troublehshoot.

The `attach()` method is used by some of the non-ASGI integrations. Look at the ASGI examples in this repository for good examples: https://github.com/miguelgrinberg/python-socketio/tree/main/examples/server/asgi
